### PR TITLE
feat(MRS): add a new parameter: public_ip

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add a new parameter: public_ip

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/mrs' TESTARGS='-run=TestAccMrsMapReduceCluster_Eip_publicIp'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/mrs -v -run=TestAccMrsMapReduceCluster_Eip_publicIp -timeout 360m -parallel 4
=== RUN   TestAccMrsMapReduceCluster_Eip_publicIp
=== PAUSE TestAccMrsMapReduceCluster_Eip_publicIp
=== CONT  TestAccMrsMapReduceCluster_Eip_publicIp
--- PASS: TestAccMrsMapReduceCluster_Eip_publicIp (808.85s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs       809.071s
```

```
make testacc TEST='./huaweicloud/services/acceptance/mrs' TESTARGS='-run=TestAccMrsMapReduceCluster_Eip_id'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/mrs -v -run=TestAccMrsMapReduceCluster_Eip_id -timeout 360m -parallel 4
=== RUN   TestAccMrsMapReduceCluster_Eip_id
=== PAUSE TestAccMrsMapReduceCluster_Eip_id
=== CONT  TestAccMrsMapReduceCluster_Eip_id
--- PASS: TestAccMrsMapReduceCluster_Eip_id (827.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs       827.340s
```

### CODE REVIEW

File: huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go [hunk 1]
```diff
@@ -131,9 +131,18 @@ func ResourceMRSClusterV2() *schema.Resource {
...
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				RequiredWith: []string{"eip_address"},
+			},
+			"eip_address": {
```
Comment (chengxiangdong) [inline huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go]: suggest using `public_ip` instead

File: huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go [hunk 2]
```diff
@@ -220,6 +220,7 @@ func TestAccMrsMapReduceCluster_stream(t *testing.T) {
 				ImportStateVerifyIgnore: []string{
 					"manager_admin_pass",
 					"node_admin_pass",
+					"eip_ip",
```
Comment (chengxiangdong) [inline huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go]: `eip_ip` is an undefined argument

File: huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go [hunk 3]
```diff
@@ -1031,3 +1032,105 @@ resource "huaweicloud_mapreduce_cluster" "test" {
...
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"manager_admin_pass",
+					"node_admin_pass",
+					"eip_id",
```
Comment (chengxiangdong) [inline huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go]: `eip_id` has been setted in read function, no need to ignore

File: docs/resources/mapreduce_cluster.md [hunk 4]
```diff
@@ -299,8 +299,11 @@ The following arguments are supported:
 * `enterprise_project_id` - (Optional, String, ForceNew) Specifies a unique ID in UUID format of enterprise project.
   Changing this will create a new MapReduce cluster resource.
 
-* `eip_id` - (Optional, String, ForceNew) Specifies the EIP ID which bound to the MapReduce cluster. Changing this will
-  create a new MapReduce cluster resource.
+* `public_ip` - (Optional, String, ForceNew) Specifies the EIP address which bound to the MapReduce cluster.
```
Comment (niuzhenguo) [inline docs/resources/mapreduce_cluster.md:360]: why users need to set both eip id and address? you can fetch the address according the provided id yoursef

**Pull request merged**